odoo/orm/models.py

```diff
@@ -741,9 +741,20 @@ def _build_model_attributes(cls, pool):
                 depends.setdefault(mname, []).extend(fnames)
 
             for cons in base._sql_constraints:
-                cls._sql_constraints[cons[0]] = cons
-
-        cls._sql_constraints = list(cls._sql_constraints.values())
+                if not isinstance(cons, DatabaseObject):
+                    # XXX this will be removed before merging
+                    cons_name, definition, *cons_message = cons
+                    match definition.strip().lower().split():
+                        case ["index", *_]:
+                            cons = Index(definition[5:].strip())
```

Comment:
If `definition` starts with a space, you are screwed.
I would use a regular expression instead, and retrieve the parts from the match.

Reply:
Yes, but it is a temporary implementation that gets overwritten in the commit after.